engine/verification/assigner/engine.go

```diff
@@ -165,6 +170,11 @@ func (e *Engine) ProcessFinalizedBlock(block *flow.Block) {
 // processFinalizedBlock indexes the execution receipts included in the block, performs chunk assignment on its result, and
 // processes the chunks assigned to this verification node by pushing them to the chunks consumer.
 func (e *Engine) processFinalizedBlock(ctx context.Context, block *flow.Block) {
+
+	if e.stopAtHeight > 0 && block.Header.Height == e.stopAtHeight {
+		e.stopAtBlockID = block.ID()
```

Comment:
I think this method is called by an external thread. Hence you might need concurrency handling here 🤔 . [Atomic Value](https://pkg.go.dev/sync/atomic#Value) might work nicely here, since you have a default initialization value (`flow.ZeroID`)

Reply:
👍